src/mito2/src/compaction/compactor.rs

```diff
@@ -0,0 +1,422 @@
+
+use crate::access_layer::{AccessLayer, AccessLayerRef, SstWriteRequest};
+use crate::cache::{CacheManager, CacheManagerRef};
+use crate::compaction::build_sst_reader;
+use crate::compaction::picker::{new_picker, PickerOutput};
+use crate::config::MitoConfig;
+use crate::error::{EmptyRegionDirSnafu, JoinSnafu, ObjectStoreNotFoundSnafu, Result};
+use crate::manifest::action::{RegionEdit, RegionMetaAction, RegionMetaActionList};
+use crate::manifest::manager::{RegionManifestManager, RegionManifestOptions};
+use crate::manifest::storage::manifest_compress_type;
+use crate::memtable::time_partition::TimePartitions;
+use crate::memtable::MemtableBuilderProvider;
+use crate::read::Source;
+use crate::region::options::RegionOptions;
+use crate::region::version::{VersionBuilder, VersionControl, VersionControlRef};
+use crate::region::ManifestContext;
+use crate::region::RegionState::Writable;
+use crate::schedule::scheduler::LocalScheduler;
+use crate::sst::file::{FileMeta, IndexType};
+use crate::sst::file_purger::{FilePurgerRef, LocalFilePurger};
+use crate::sst::index::intermediate::IntermediateManager;
+use crate::sst::parquet::WriteOptions;
+
+/// CompactionRegion represents a region that needs to be compacted.
+/// It's the subset of MitoRegion.
+#[derive(Clone)]
+pub struct CompactionRegion {
+    pub region_id: RegionId,
+    pub region_options: RegionOptions,
+
+    pub(crate) engine_config: Arc<MitoConfig>,
+    pub(crate) region_metadata: RegionMetadataRef,
+    pub(crate) cache_manager: CacheManagerRef,
+    pub(crate) access_layer: AccessLayerRef,
+    pub(crate) file_purger: FilePurgerRef,
+    pub(crate) manifest_ctx: Arc<ManifestContext>,
+    pub(crate) version_control: VersionControlRef,
+}
+
+/// CompactionRequest represents the request to compact a region.
+#[derive(Debug, Clone)]
+pub struct CompactionRequest {
+    pub catalog: String,
+    pub schema: String,
+    pub region_id: RegionId,
+    pub options: HashMap<String, String>,
+    pub compaction_options: compact_request::Options,
+}
+
+/// Open a compaction region from a compaction request.
+/// It's simple version of RegionOpener::open().
+pub async fn open_compaction_region(
+    req: &CompactionRequest,
+    data_home: &str,
+    object_store_manager: ObjectStoreManager,
+) -> Result<CompactionRegion> {
+    let region_options = RegionOptions::try_from(&req.options)?;
+    let region_dir = region_dir(
+        format!("{}/{}", &req.catalog, &req.schema).as_str(),
+        req.region_id,
+    );
+
+    let object_store = {
+        let name = &region_options.storage;
+        if let Some(name) = name {
+            object_store_manager
+                .find(name)
+                .context(ObjectStoreNotFoundSnafu {
+                    object_store: name.to_string(),
+                })?
+        } else {
+            object_store_manager.default_object_store()
+        }
+    };
+
+    let mut mito_config = MitoConfig::default();
+    mito_config.sanitize(data_home)?;
```

Comment:
It'd be better to pass the mito_config to this function. Then it doesn't need to care about the `data_home`.